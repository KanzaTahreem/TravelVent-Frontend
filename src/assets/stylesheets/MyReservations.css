.my-reservations {
  padding: 20px;
  margin-top: 10px;
  gap: 10px;
}
