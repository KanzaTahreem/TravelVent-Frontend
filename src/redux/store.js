import { configureStore } from '@reduxjs/toolkit';
import cartItems from './cartSlice';
import registrationSlice from './reducers/registrationSlice';

const store = configureStore({
  reducer: {
    cartItems,
    registration: registrationSlice,
  },
});

export default store;
