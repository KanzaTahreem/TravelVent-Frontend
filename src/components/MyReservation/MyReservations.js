import ReservationItem from './ReservationItem';
import './MyReservations.css';

export default function MyReservations() {
  return (
    <ul className="my-reservations column">
      <ReservationItem />
      <ReservationItem />
      <ReservationItem />
    </ul>
  );
}
