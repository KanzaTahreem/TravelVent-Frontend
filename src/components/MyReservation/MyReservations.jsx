import { useDispatch, useSelector } from 'react-redux';
import { useEffect } from 'react';
import { getAllReversedTrips } from '../../redux/reserveSlice';
import ReservationItem from './ReservationItem';
import './MyReservations.css';

export default function MyReservations() {
  const reservations = useSelector((store) => store.reserveItems.reserved);
  const dispatch = useDispatch();

  useEffect(() => {
    dispatch(getAllReversedTrips());
  }, [dispatch]);

  return (
    <ul className="my-reservations column">
      {reservations.map((reservation, index) => (
        <ReservationItem key={reservation.id} reservation={reservation} delay={`1.${index}`} />
      ))}
    </ul>
  );
}
