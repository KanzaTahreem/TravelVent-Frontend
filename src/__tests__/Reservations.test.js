import '@testing-library/jest-dom';
import { BrowserRouter as Router } from 'react-router-dom';
import renderer from 'react-test-renderer';
import { screen, waitFor } from '@testing-library/react';
import { Provider } from 'react-redux';
import store from '../redux/store';
import MyReservations from '../pages/MyResevationsPage';

describe('RootLayout', () => {
  let myReservation;
  beforeAll(async () => {
    myReservation = renderer
      .create(
        <Provider store={store}>
          <Router>
            <MyReservations />
          </Router>
        </Provider>,
      )
      .toJSON();
  });

  it('should renders correctly', () => {
    expect(myReservation).toBeTruthy();
  });

  it('should match the snapshot', () => {
    expect(myReservation).toMatchSnapshot();
  });

  test('renders the component with the correct heading', () => {
    const heading = waitFor(() => screen.getByRole('heading', { name: 'Adventure Awaits' }));
    waitFor(() => expect(heading).toBeInTheDocument());
  });
});
