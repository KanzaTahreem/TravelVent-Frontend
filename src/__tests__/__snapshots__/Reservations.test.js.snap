// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`RootLayout should match the snapshot 1`] = `
<div
  className="my-reservations-div"
>
  <h1>
    Adventure Awaits
  </h1>
  <p>
    Get a list of your reservations!
  </p>
  <ul
    className="my-reservations column"
  />
</div>
`;
